pkg/pushsync/pushsync_test.go

```diff
@@ -83,148 +73,191 @@ func TestSendToClosest(t *testing.T) {
 		t.Fatal("too many messages")
 	}
 	delivery := messages[0].(*pb.Delivery)
-	chunk := swarm.NewChunk(swarm.NewAddress(delivery.Address), delivery.Data)
+	rcvdChunk := swarm.NewChunk(swarm.NewAddress(delivery.Address), delivery.Data)
 
-	if !bytes.Equal(chunk.Address().Bytes(), chunkAddress.Bytes()) {
+	if !bytes.Equal(rcvdChunk.Address().Bytes(), chunkAddress.Bytes()) {
 		t.Fatalf("chunk address mismatch")
 	}
 
-	if !bytes.Equal(chunk.Data(), chunkData) {
+	if !bytes.Equal(rcvdChunk.Data(), chunkData) {
 		t.Fatalf("chunk data mismatch")
 	}
-}
 
-// TestForwardChunk tests that when a closer node exists within the topology, we forward a received
-// chunk to it.
-func TestForwardChunk(t *testing.T) {
-	logger := logging.New(ioutil.Discard, 0)
+	// this intercepts the incoming receipt message
+	records = recorder.WaitRecords(t, closestPeer, pushsync.ProtocolName, pushsync.ProtocolVersion, pushsync.StreamName, 1, 5)
+	messages, err = protobuf.ReadMessages(
+		bytes.NewReader(records[0].In()),
+		func() protobuf.Message { return new(pb.Receipt) },
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(messages) > 1 {
+		t.Fatal("too many messages")
+	}
+	receipt := messages[0].(*pb.Receipt)
+	receiptAddress := swarm.NewAddress(receipt.Address)
+
+	if !receiptAddress.Equal(chunkAddress) {
+		t.Fatalf("receipt address mismatch")
+	}
+}
 
+// TestHandler expect a chunk from a node on a stream. It then stores the chunk in the local store and
+// sends back a receipt. This is tested by intercepting the incoming stream for proper messages.
+// It also sends the chunk to the closest peerand receives a receipt.
+//
+// Chunk moves from   TriggerPeer -> PivotPeer -> ClosestPeer
+//
+func TestHandler(t *testing.T) {
 	// chunk data to upload
 	chunkAddress := swarm.MustParseHexAddress("7000000000000000000000000000000000000000000000000000000000000000")
 	chunkData := []byte("1234")
+	chunk := swarm.NewChunk(chunkAddress, chunkData)
 
-	// create a pivot node and a closest mocked closer node address
-	pivotNode := swarm.MustParseHexAddress("0000000000000000000000000000000000000000000000000000000000000000")   // pivot is 0000
-	closestPeer := swarm.MustParseHexAddress("6000000000000000000000000000000000000000000000000000000000000000") // binary 0110
-
-	// Create a mock connectivity driver
-	mockTopology := mock.NewTopologyDriver(mock.WithClosestPeer(closestPeer))
-	storer, err := localstore.New("", pivotNode.Bytes(), nil, logger)
-	if err != nil {
-		t.Fatal(err)
-	}
+	// create a pivot node and a mocked closest node
+	pivotPeer := swarm.MustParseHexAddress("0000000000000000000000000000000000000000000000000000000000000000")
+	triggerPeer := swarm.MustParseHexAddress("6000000000000000000000000000000000000000000000000000000000000000")
+	closestPeer := swarm.MustParseHexAddress("6000000000000000000000000000000000000000000000000000000000000000")
```

Comment:
triggerPeer and closestPeer have the same address and this is confusing, and could potentially eclipse other problems in the tests